### PR TITLE
Add balanced test split

### PR DESCRIPTION
**Pull request by thomkeh**

### CODE REVIEW

File: ethicml/preprocessing/train_test_split.py [hunk 1]
```diff
@@ -237,6 +238,77 @@ def __call__(
...
+            random.shuffle(idx)
+            split_indexes: int = round(len(idx) * self.train_percentage)
+            # append index subsets to the list of train indices
+            train_indexes.append(idx[:split_indexes])
+            test_indexes.append(idx[split_indexes : (split_indexes + num_test_balanced[y])])
+            num_dropped += num_test[(s, y)] - num_test_balanced[y]
+
+        train_idx = np.concatenate(train_indexes, axis=0)
+        test_idx = np.concatenate(test_indexes, axis=0)
+        del train_indexes
```
Comment (olliethomas) [inline ethicml/preprocessing/train_test_split.py]: I'm not sure these `del` statements are worth it

Comment (thomkeh) [inline ethicml/preprocessing/train_test_split.py]:
```suggestion
```

File: ethicml/preprocessing/train_test_split.py [hunk 2]
```diff
@@ -237,6 +238,77 @@ def __call__(
...
+        del test_indexes
```
Comment (thomkeh) [inline ethicml/preprocessing/train_test_split.py]:
```suggestion
```

File: ethicml/preprocessing/train_test_split.py (hunk 1 above)
Comment (thomkeh) [reply ethicml/preprocessing/train_test_split.py]: removed!

**Pull request merged**